### PR TITLE
[ASPagerNode] Fix ASPagerNode data source and delegate inconsistencies

### PR DESCRIPTION
**Pull request by maicki**
- Change ASPagerNodeDataSource -> ASPagerDataSource
- Add ASPagerDelegate

### CODE REVIEW

File: AsyncDisplayKit/ASPagerNode.h [hunk 1]
```diff
@@ -65,25 +66,30 @@
 
 @end
 
+@protocol ASPagerDelegate <NSObject, ASCollectionDelegate>
```
Comment (appleguy) [inline AsyncDisplayKit/ASPagerNode.h]: This doesn't need NSObject if the one above doesn't, right?

Comment (maicki) [inline AsyncDisplayKit/ASPagerNode.h]: Yep, ASCollectionDelegate already extends it

**Comment by maicki** [comment]
@appleguy Addressed your comments

**Comment by levi** [comment]
LGTM

**Pull request merged**